lib/plugins/aws/customResources/resources/utils.js

```diff
@@ -55,22 +55,34 @@ function response(event, context, status, data = {}, err) {
   });
 }
 
-function getLambdaArn(region, accountId, functionName) {
-  return `arn:aws:lambda:${region}:${accountId}:function:${functionName}`;
+function getLambdaArn(partition, region, accountId, functionName) {
+  return `arn:${partition}:lambda:${region}:${accountId}:function:${functionName}`;
 }
 
 function getEnvironment(context) {
   const arn = context.invokedFunctionArn.match(
-    /^arn:aws.*:lambda:(\w+-\w+-\d+):(\d+):function:(.*)$/
+    /^arn:(aws[\w-]*).*:lambda:([\w+-]{2,}\d+):(\d+):function:(.*)$/
   );
+
   return {
     LambdaArn: arn[0],
-    Region: arn[1],
-    AccountId: arn[2],
-    LambdaName: arn[3],
+    Partition: arn[1],
+    Region: arn[2],
+    AccountId: arn[3],
+    LambdaName: arn[4],
   };
 }
 
+function getPartition(region) {
```

Comment:
I think this function should not be introduced. Instead in all places where it's used, we should take _partition_ in arguments together with a _region_ (and both via `getEnvironment` hould be resolved in handler context)

Reply:
@medikoo , I'm not exactly sure what you you would like changed here?  I am trying to calculate Partition from a given region which I discovered was incorrectly being parsed.  This function is used in 3 locations, where Partition was being parsed from Region to form a non-lambda ARN.  See the three locations here, which were not properly handling GovCloud:
line 17  - lib/plugins/aws/customResources/resources/cognitoUserPool/lib/permissions.js
line 17 - lib/plugins/aws/customResources/resources/eventBridge/lib/permissions.js
line 17 - lib/plugins/aws/customResources/resources/s3/lib/permissions.js

I wanted to ensure that all use-cases for Custom Resources are now properly handled by the framework for GovCloud, which this solves.  

Thank you.

Reply:
After looking thru, I may get what you are saying, please correct me if i'm wrong.  You are suggesting that I pass Partition into the permissions.js addPermission() / removePermission() functions, after getting the Partition in the handler.js functions, via getEnvironment?  Sorry, I just want to understand what you are asking before I pursue the changes.

Reply:
>  You are suggesting that I pass Partition into the permissions.js addPermission() / removePermission() functions,

Yes, exactly, sorry, for not being perfectly clear